src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonService.kt

```diff
@@ -3,64 +3,50 @@ package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+  private fun getPrisonApiSourceData(prisonerDetails: PrisonerDetails, activeDataOnly: Boolean = true): PrisonApiSourceData {
+    val sentenceAndOffences = getSentencesAndOffences(prisonerDetails.bookingId, activeDataOnly)
+    val bookingAndSentenceAdjustments = getBookingAndSentenceAdjustments(prisonerDetails.bookingId, activeDataOnly)
+    val bookingHasFixedTermRecall = sentenceAndOffences.any { isSupported(it.sentenceCalculationType) && from(it.sentenceCalculationType).recallType?.isFixedTermRecall == true }
+    val (ftrDetails, returnToCustodyDate) = getFixedTermRecallDetails(prisonerDetails.bookingId, bookingHasFixedTermRecall)
+    val bookingHasAFine = sentenceAndOffences.any { isSupported(it.sentenceCalculationType) && from(it.sentenceCalculationType).sentenceClazz == AFineSentence::class.java }
+    val offenderFinePayments = if (bookingHasAFine) prisonApiClient.getOffenderFinePayments(prisonerDetails.bookingId) else listOf()
+
+    return PrisonApiSourceData(
+      sentenceAndOffences,
+      prisonerDetails,
+      bookingAndSentenceAdjustments,
+      offenderFinePayments,
+      returnToCustodyDate,
+      ftrDetails
+    )
   }
 
-  private fun getInactivePrisonApiSourceData(prisonerDetails: PrisonerDetails): PrisonApiSourceData {
-    val sentenceAndOffences = getSentencesAndOffences(prisonerDetails.bookingId, false)
-    val bookingAndSentenceAdjustments = getBookingAndSentenceAdjustments(prisonerDetails.bookingId, false)
-    val bookingHasFixedTermRecall = sentenceAndOffences.any { SentenceCalculationType.from(it.sentenceCalculationType)?.recallType?.isFixedTermRecall == true }
-    var returnToCustodyDate: ReturnToCustodyDate? = null
-    if (bookingHasFixedTermRecall) {
-      // TODO look into replacing ReturnToCustodyDate object with FixedTermRecallDetails
-      returnToCustodyDate = transform(prisonApiClient.getFixedTermRecallDetails(prisonerDetails.bookingId))
-    }
-    val bookingHasAFine = sentenceAndOffences.any { SentenceCalculationType.from(it.sentenceCalculationType)?.sentenceClazz == AFineSentence::class.java }
-    var offenderFinePayments: List<OffenderFinePayment> = listOf()
-    if (bookingHasAFine) {
-      offenderFinePayments = prisonApiClient.getOffenderFinePayments(prisonerDetails.bookingId)
-    }
-    return PrisonApiSourceData(sentenceAndOffences, prisonerDetails, bookingAndSentenceAdjustments, offenderFinePayments, returnToCustodyDate)
+  private fun getFixedTermRecallDetails(bookingId: Long, bookingHasFixedTermRecall: Boolean): Pair<FixedTermRecallDetails?, ReturnToCustodyDate?> {
+    // TODO Remove ReturnToCustodyDate - FixedTermRecallDetails is all that's required. Will be done as tech debt
+    if (!bookingHasFixedTermRecall) return Pair(null, null)
+    val ftrDetails = prisonApiClient.getFixedTermRecallDetails(bookingId)
+    val returnToCustodyDate = transform(ftrDetails)
+    return Pair(ftrDetails, returnToCustodyDate)
```

Comment:
I think you can do `ftrDetails to returnToCustodyDate` to return a pair

Reply:
good spot - done